### PR TITLE
AP_Scripting: add ability to load scripts as librarys

### PR DESCRIPTION
**Pull request by IamPete1**
This sort of gets us `requires` support. We can load a script that returns a table of the functions and variables it has, we can then call the functions and scripts from the 'master' script.

Having to put all the stuff you need inside the lib in a table it abit annoying, not sure how to do it better tho.

### CODE REVIEW

File: libraries/AP_Scripting/lua_bindings.cpp [hunk 1]
```diff
@@ -75,11 +75,58 @@ static int lua_mission_receive(lua_State *L) {
...
+    lua_pushstring(L, "utf8");
+    luaopen_utf8(L);
+    lua_settable(L, -3);
+    load_lua_bindings(L);
+    load_generated_sandbox(L);
+    lua_setupvalue(L, -2, 1);
+
+    // run the libary, it should return a table to its varables and functions
+    lua_pcall(L, 0, 1, 0);
+
```
Comment (IamPete1) [inline libraries/AP_Scripting/lua_bindings.cpp:120]: we could do some better error handling here

**Comment by WickedShell** [comment]
I haven't looked into it, but why is requires hard to support correctly? It
should be fairly straightforward I thought....

**Comment by IamPete1** [comment]
@WickedShell I did look at doing it properly but completely failed to get it working.

**Comment by IamPete1** [comment]
I used this to load a matrix library, it worked great. https://github.com/davidm/lua-matrix/blob/master/lua/matrix.lua

**Comment by IamPete1** [comment]
Superseded by https://github.com/ArduPilot/ardupilot/pull/23354

**Pull request closed without merging**